src/goto-instrument/contracts/contracts.cpp

```diff
@@ -473,18 +473,19 @@ code_contractst::create_ensures_instruction(
 }
 
 bool code_contractst::apply_function_contract(
-  goto_programt &goto_program,
-  goto_programt::targett target)
+  const irep_idt &function,
+  goto_programt &function_body,
+  goto_programt::targett &target)
 {
   // Return if the function is not named in the call; currently we don't handle
   // function pointers.
-  PRECONDITION(as_const(*target).call_function().id() == ID_symbol);
+  PRECONDITION((*target).call_function().id() == ID_symbol);
```

Comment:
Why are we removing `as_const`?

Reply:
It's unnecessary.

Reply:
Why?
It returns a `const` reference, so we wouldn't be able to unintentionally modify the underlying object.

For example, in this case, we were grabbing a `const` ref to `*target` and then calling `call_function()` on it, which previously wouldn't have been able to change `*target`, but now it could.

Reply:
@SaswatPadhi but now the code is
```c
PRECONDITION(target->call_function().id() == ID_symbol);
```

Reply:
Yes, but the `target->call_function()` is no longer guaranteed not to change `*target`.